### PR TITLE
Update to API version `2020-06-01` for App Service

### PR DESCRIPTION
**Pull request by jackofallops**
Address Breaking changes: 
- [x] Enum for `internal_load_balancing_mode` changed from `Web, Publishing` to `Web,Publishing`

### CODE REVIEW

**Comment by AdamCoulterOz** [comment]
> Breaking changes:
> 
> * Enum for `internal_load_balancing_mode` changed from `Web, Publishing` to `Web,Publishing`

Could it not be made non-breaking by adding:

```go
internalLoadBalancingMode := d.Get("internal_load_balancing_mode").(string)
internalLoadBalancingMode = strings.ReplaceAll(internalLoadBalancingMode, " ", "")
```

and adding a warning?

**Comment by tombuildsstuff** [comment]
+1 we can add a bi-directional mapping here - converting to/from `Web, Publishing` - since other API's use the same `[value1], [value2]` convention for MSI

File: azurerm/internal/services/web/app_service.go [hunk 1]
```diff
@@ -91,6 +91,32 @@ func schemaAppServiceGoogleAuthSettings() *schema.Schema {
...
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"client_id": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
```
Comment (katbyte) [inline azurerm/internal/services/web/app_service.go:104]: Should we be validating this?

File: azurerm/internal/services/web/app_service.go [hunk 2]
```diff
@@ -91,6 +91,32 @@ func schemaAppServiceGoogleAuthSettings() *schema.Schema {
...
+				"client_secret": {
+					Type:      schema.TypeString,
+					Required:  true,
+					Sensitive: true,
+				},
```
Comment (katbyte) [inline azurerm/internal/services/web/app_service.go:109]: and this? (with at least no empty?)

File: azurerm/internal/services/web/app_service.go [hunk 3]
```diff
@@ -91,6 +91,32 @@ func schemaAppServiceGoogleAuthSettings() *schema.Schema {
...
+				"oauth_scopes": {
+					Type:     schema.TypeList,
+					Optional: true,
+					Elem:     &schema.Schema{Type: schema.TypeString},
```
Comment (katbyte) [inline azurerm/internal/services/web/app_service.go:113]: and can this be validated too?

**Review by katbyte — COMMENTED** [review]
Aside from a few comments this LGTM :

**Review by katbyte — APPROVED** [review]

File: azurerm/internal/services/web/app_service.go (hunk 1 above)
Comment (jackofallops) [reply azurerm/internal/services/web/app_service.go]: pulled this in favour of adding as a separate PR.

File: azurerm/internal/services/web/app_service.go (hunk 3 above)
Comment (jackofallops) [reply azurerm/internal/services/web/app_service.go]: pulled this in favour of adding as a separate PR.

File: azurerm/internal/services/web/app_service.go (hunk 2 above)
Comment (jackofallops) [reply azurerm/internal/services/web/app_service.go]: pulled this in favour of adding as a separate PR.

**Comment by jackofallops** [comment]
Tests looking good, (failures unrelated to SDK change)
image

**Pull request merged**